app/org/sagebionetworks/bridge/BridgeUtils.java

```diff
@@ -57,10 +57,17 @@ public static String generateGuid() {
         return UUID.randomUUID().toString();
     }
     
-    public static String generateTaskRunKey(Task task) {
-        checkNotNull(task.getSchedulePlanGuid());
-        checkNotNull(task.getScheduledOn());
-        return String.format("%s:%s", task.getSchedulePlanGuid(), Long.toString(task.getScheduledOn()));
+    /**
+     * Identifies a set of tasks from a single run of a schedule. 
+     * @param task
+     * @return
+     */
+    public static String generateTaskRunKey(DynamoTask task, ScheduleContext context) {
```

Comment:
Why are we using DynamoTask instead of Task? Why couple it tightly to the Dynamo-specific implementation?